### DISCUSSION TITLE
where is the Julia code used to produce the movies in README.md?

### DISCUSSION THREAD

**Discussion by waywardpidgeon** [CliMA/Oceananigans.jl#3839] [General]
I am exploring Oceananigans.jl examples and would like to try and reproduce the examples of the movies in README.md. Is the code somewhere I can use or modify?
Thanks

**Comment by ali-ramadhan** [comment]
Hi @waywardpidgeon!

Are you interested in the simulation setup code or the plotting code?

Unfortunately, the movies in the README.md were generated using ancient versions of Oceananigans. So any code we might unearth from git history would be way too old to work with the current version of Oceananigans. That said, I'm happy to share the details so you could reproduce them.

They're actually simple simulations to set up. Just a fluid at rest forced by surface buoyancy and/or momentum fluxes. In the first movie (deep convection) the surface buoyancy flux is restricted to a circle of some radius. I think all simulations in those movies used 256^3 grid points. Today Oceananigans has much better advection schemes so you could produce much higher fidelity simulations at the same resolution!

The plots were made using matplotlib (way before Julia had Makie.jl) so if I can find that code, it could be useful. But if I were to remake the plots today I'd use Makie.jl (and GLMakie.jl for 3D plots).

**Reply by glwagner** [reply]
3D plots are demonstrated in the baroclinic adjustment script: https://github.com/CliMA/Oceananigans.jl/blob/main/examples/baroclinic_adjustment.jl

Some simplifications would be warranted for a movie like the one in the README. There is also an example of an animation in that script.

Unfortunately, it is too expensive to make 3D movies in examples right now, so we don't have a way of continuously testing such code.

That said it would not be hard to produce something in this discussion. I have a lot of similar code lying around. We just need to know whether the objective is simulating or movie-making or both.

**Comment by ali-ramadhan** [comment]
So I actually found the script used for the deep convection simulations from way back in v0.5.3! You'll notice the code looks much more ragged and completely different to today's interface but the parameter values are there if you want to recreate the exact same setup: https://github.com/CliMA/Oceananigans.jl/blob/v0.5.3/examples/deep_convection.jl

And I believe I used matplotlib's `contour3d` to make the plots: https://github.com/CliMA/Oceananigans.jl/blob/v0.5.3/examples/plot_oceananigans_output_contourf3d.py (although there's also a version of the script in there that uses `pcolormesh`).

**Comment by waywardpidgeon** [comment]
Oh, thanks for that. Please don't spend any time on this  - it's far from essential.
Cheers Kevin

________________________________
